unidist/run.py

```diff
@@ -0,0 +1,247 @@
+        Name of .py script to be run.
+    executor : str, default: 'python3'
+        Executable to run `script`.
+    backend : str, default: 'Ray'
+        Unidist backend name to use.
+    num_workers : list, default: ['localhost' cpu count]
+        List of string values. The each value represents
+        a number of workers for corresponded host.
+    hosts : list, default: ['localhost']
+        List of strings with ip-addresses.
+
+    Returns
+    -------
+    list
+        List of strings represents command for ``subprocess``.
+    """
+    if backend == "MPI":
+        unidist_home_path = _get_unidist_home_path()
+
+        if len(hosts) != len(num_workers):
+            # If `num_workers` isn't provided or a single value `default` is provided
+            # all workers will use `default` value
+            if len(num_workers) == 1 and num_workers[0] == "default":
+                num_workers *= len(hosts)
+            else:
+                raise RuntimeError(
+                    "`num_workers` and `hosts` parameters must have the equal number of values."
+                )
+
+        hosts = _validate_hosts(hosts)
+        num_workers = _validate_num_workers(num_workers)
+        workers_dir = "/tmp"
+        command = ["mpiexec", "-hosts"]
+
+        hosts_str = f"{_get_localhost_ip()}:1,{_get_localhost_ip()}:1,"
+        for host, n in zip(hosts, num_workers):
+            hosts_str += host + ":" + n + ","
+
+        command_executor = ["-n", "1", executor, script]
+        command_monitor = [
+            "-n",
+            "1",
+            "-wdir",
+            workers_dir,
+            "python3",
+            unidist_home_path + "/unidist/core/backends/mpi/core/monitor.py",
+        ]
+
+        def get_worker_command(num_workers):
+            return [
+                "-n",
+                num_workers,
+                "-wdir",
+                workers_dir,
+                "python3",
+                unidist_home_path + "/unidist/core/backends/mpi/core/worker.py",
+            ]
+
+        command += [hosts_str] + command_executor + [":"] + command_monitor
+
+        for n in num_workers:
+            command += [":"] + get_worker_command(n)
+    else:
+        command = [executor, script]
+
+    return command
+
+
+def main():
+    """Run unidist application."""
+    usage_examples = [
+        "\n\tIn case 'unidist' is installed, use binary:",
+        "\n\tunidist script.py  # Ray backend is used",
+        "\n\tunidist script.py --backend MPI  # MPI backend is used",
+        "\n\tunidist script.py --executor pytest -b Dask  # Dask backend is used, running using 'pytest'",
+        "\n\n\tTo run from sources use 'unidist/run.py':",
+        "\n\tpython3 unidist/run.py script.py -b MPI --num_workers=16 --hosts localhost  # MPI backend uses 16 workers on 'localhost' node",
+        "\n\tunidist script.py -b MPI --num_workers=2 4 --hosts localhost x.x.x.x  # MPI backend uses 2 workers on 'localhost' and 4 on 'x.x.x.x'",
```

Comment:
Btw, specifying values for `num_workers` and `hosts` would be good to generalize (use the same setting values).

Reply:
Suggestion was applied.



> Btw, specifying values for `num_workers` and `hosts` would be good to generalize (use the same setting values).

Do you mean something like that? 
```bash
user$: unidist script.py -b MPI -hosts localhost:4 192.168.1.45:16
```
In other case please describe what do you mean.

Reply:
Help lines were fixed. I suggest to leave `num_workers` and `num_hosts` as it is by the next reasons:
1) It would be more flexible to combine separate parameters for different backends by different ways.
2) `num_workers` will be renamed to `num_cpus`. Possible, we will add others options like `num_gpus` in the future. Separate parameters will be more clear for users I think.